packages/medusa/src/services/discount.js

```diff
@@ -524,6 +532,23 @@ class DiscountService extends BaseService {
     })
   }
 
+  /**
+   * Increments the usage of a discount code
+   * @param {string} discountId - id of discount
+   * @return {Promise} the result of the update operation
+   */
+  async incrementUsageCount(discountId) {
+    return this.atomicPhase_(async (manager) => {
+      const discountRepo = manager.getCustomRepository(this.discountRepository_)
+
+      const discount = await this.retrieve(discountId)
```

Comment:
Instead of retrieving the data and then update it, why dont just updating it?

```
await this.manager_.createQueryBuilder()
	.update(Discount)
	.set({ usage_count: () => "coalesce(usage_count, 0) + 1" })
	.where("id = :id", { id: discountId })
.returning("*")
.execute()
```

To avoid having to coalesce, the column should have a default value set to 0.